FinalTest/graph.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "graph.h"
+
+
+typedef struct Graph  {
+    int numNodes;
+    bool** edges;
+} Graph;
+
+Graph* createGraph(int numNodes) {
+    Graph *graph = malloc(sizeof(Graph));
+    if (graph == NULL) {
+        return NULL;
+    }
+
+    graph->numNodes = numNodes;
+
+    graph->edges = calloc(sizeof(bool*), graph->numNodes);
+    if (graph->edges == NULL) {
+        free(graph);
+        return NULL;
+    }
+
+    for (int i = 0; i < graph->numNodes; i++) {
+        graph->edges[i] = calloc(sizeof(bool), graph->numNodes);
+        if (graph->edges[i] == NULL) {
+            deleteGraph(graph);
+            return NULL;
+        }
+    }
+
+    for (int i = 0; i < numNodes; i++) {
+        graph->edges[i][i] = true;
+    }
+
+    return graph;
+}
+
+void deleteGraph(Graph* graph) {
+    if (graph->edges == NULL) {
+        return;
+    }
+    for (int i = 0; i < graph->numNodes; i++) {
+        if (graph->edges[i] != NULL) {
+            free(graph->edges[i]);
+        }
```

Comment:
```suggestion
        free(graph->edges[i]);
```
Это то же самое, free по стандарту ничего не делает, если ей передали NULL